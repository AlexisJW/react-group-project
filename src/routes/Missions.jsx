import React from 'react';

const Missions = () => (
  <main className="missions-main">
    <section className="missions-section">Missions</section>
  </main>
);

export default Missions;
