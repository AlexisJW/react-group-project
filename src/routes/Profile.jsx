import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { getMissionFromApi } from '../redux/missions/missionsSlice';

const Profile = () => {
  const dispatch = useDispatch();
  const {
    missions: { missions },
    rockets: { rockets },
  } = useSelector((state) => state);

  useEffect(() => {
    if (missions.length === 0) {
      dispatch(getMissionFromApi());
    }
  }, [missions, dispatch]);

  const filteringMission = missions.filter((mission) => mission.joined === true);

  return (
    <main className="profile-main">
      <section className="profile-mission">
        <h2> My Missions </h2>
        <table>
          <tbody>
            {filteringMission.length > 0 ? (
              <>
                {missions.filter((mission) => mission.joined).map((mission) => (
                  <tr key={mission.mission_id}>
                    <td>
                      {mission.mission_name}
                    </td>
                  </tr>
                ))}
              </>
            ) : (
              <tr>
                <td> No joined missions </td>
              </tr>
            )}
          </tbody>
        </table>
      </section>
      <section className="profile-rockets">
        <h2> My Rockets </h2>
        <table>
          <tbody>
            {rockets.length > 0 ? (
              <>
                {rockets.filter((rocket) => rocket.isReserved).map((rocket) => (
                  <tr key={rocket.id}>
                    <td>
                      {rocket.name}
                    </td>
                  </tr>
                ))}
              </>
            ) : (
              <tr>
                {' '}
                <td> No Reserved Rockets </td>
                {' '}
              </tr>
            )}
          </tbody>
        </table>
      </section>
    </main>
  );
};

export default Profile;
