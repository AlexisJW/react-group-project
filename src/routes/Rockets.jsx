import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { getRocketsFromApi } from '../redux/rockets/rocketsSlice';

const Rockets = () => {
  const dispatch = useDispatch();
  const rockets = useSelector((state) => state.rockets.rockets);

  useEffect(() => {
    if (rockets.length === 0) {
      dispatch(getRocketsFromApi());
    }
  }, [rockets, dispatch]);

  return (
    <main className="rockets-main">
      <section className="rockets-section"> Rockets </section>
    </main>
  );
};

export default Rockets;
