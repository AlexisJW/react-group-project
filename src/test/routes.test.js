import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { configureStore } from '@reduxjs/toolkit';
import missionsReducer from '../redux/missions/missionsSlice';
import rocketsReducer from '../redux/rockets/rocketsSlice';
import Missions from '../routes/Missions';
import Rockets from '../routes/Rockets';
import Profile from '../routes/Profile';

jest.mock('axios');
describe('Test for routes', () => {
  const store = configureStore({
    reducer: {
      missions: missionsReducer,
      rockets: rocketsReducer,
    },
  });

  it('Missions renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Missions />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Profile renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Profile />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Rockets renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Rockets />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
