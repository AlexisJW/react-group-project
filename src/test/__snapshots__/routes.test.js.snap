// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for routes Missions renders correctly 1`] = `
<main
  className="mission-main"
>
  <section
    className="mission-section"
  >
    <table>
      <thead
        className="text-xs text-gray-700 uppercase bg-white"
      >
        <tr>
          <th
            className="py-3 px-6 border"
            scope="col"
          >
            Mission
          </th>
          <th
            className="py-3 px-6 border"
            scope="col"
          >
            Description
          </th>
          <th
            className="py-3 px-6 border"
            scope="col"
          >
            Status
          </th>
          <th
            className="py-3 border"
            scope="col"
          >
             
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </section>
</main>
`;

exports[`Test for routes Profile renders correctly 1`] = `
<main
  className="profile-main"
>
  <section
    className="profile-mission"
  >
    <h2>
       My Missions 
    </h2>
    <table>
      <tbody>
        <tr>
          <td>
             No joined missions 
          </td>
        </tr>
      </tbody>
    </table>
  </section>
  <section
    className="profile-rockets"
  >
    <h2>
       My Rockets 
    </h2>
    <table>
      <tbody>
        <tr>
           
          <td>
             No Reserved Rockets 
          </td>
           
        </tr>
      </tbody>
    </table>
  </section>
</main>
`;

exports[`Test for routes Rockets renders correctly 1`] = `
<main
  className="rockets-main"
>
  <section
    className="rockets-section"
  />
</main>
`;
